qcodes/instrument_drivers/mightex/SSE1304U.py

```diff
@@ -0,0 +1,196 @@
+                           label='Intensity')
+
+        self.add_parameter('averaging',
+                           get_cmd=self.get_averaging,
+                           set_cmd=self.set_averaging,
+                           vals=vals.Ints(1))
+
+        self.add_parameter('darkframe',
+                           get_cmd=self.rec_darkframe,
+                           set_cmd=self.set_darkframe,
+                           vals=vals.Arrays(shape=(MightexSSE_1304_U.PIXELS,)))
+
+        self.connect()
+
+        self.drkdata = np.zeros((MightexSSE_1304_U.PIXELS,))
+
+    def connect(self):
+        """ Connects to the spectrometer and initializes the dll. """
+        self.free_device()  # there might be leftovers from previous runs
+
+        n_connected = self._dll.init(None)
+        if n_connected < self._dev_id:
+            raise RuntimeError(f'No Mightex with id {self._dev_id} connected')
+
+    def disconnect(self):
+        """ Releases all resources for the spectrometer. """
+        self._dll.uninit()
+
+    def __del__(self):
+        self.free_device()
+
+    def set_exp_t(self, t: float) -> None:
+        """ Sets the exposure time to `t` seconds. """
+        # The spectrometer can only handle multiples of 100 microseconds
+        self._exp_t = int(round(t * 1e4) * 100)
+        self._dll.set_exposuretime(self._dev_id, self._exp_t)
+
+    def get_exp_t(self) -> float:
+        return self._exp_t / 1e6  # in seconds
+
+    def set_onoff(self, on: bool) -> None:
+        """ Turns the device on and off. """
+        self._is_on = on
+        self._dll.set_activestatus(self._dev_id, 1 if on else 0)
+
+    def is_on(self) -> bool:
+        return self._is_on
+
+    def get_serialno(self) -> str:
+        """ Retrieves the device's serial number. """
+        mod_no = ct.create_string_buffer(b'\000' * 32)
+        ser_no = ct.create_string_buffer(b'\000' * 32)
+        self._dll.get_serialno(self._dev_id, mod_no, ser_no)
+        return mod_no.value.decode() + ':' + ser_no.value.decode()
+
+    def set_auto_drk(self, on: bool) -> None:
+        """ Sets auto dark compensation for the spectrometer. """
+        self._auto_drk_on = on
+        self._dll.set_auto_dark(self._dev_id, 1, 1 if on else 0)
+
+    def is_auto_drk(self) -> bool:
+        return self._auto_drk_on
+
+    def get_single_frame(self) -> np.ndarray:
+        """ Retrieves a raw single frame from the spectrometer. """
+        buffer = ct.POINTER(FrameRecord)()
+
+        self._dll.start_framegrab(self._dev_id)
+        self._dll.get_framedata(self._dev_id, 1, 1, ct.byref(buffer))
+
+        return npct.as_array(buffer.contents.CCDData,
+                             shape=(MightexSSE_1304_U.PIXELS,)).copy()
+
+    @property
+    def calib(self):
+        """ Returns a function to calibrate pixels to wavelength. """
+        # These values are taken from the default calibration file
+        # shipped with the device.
+        return np.poly1d([
```

Comment:
What happens when we use this driver on another instance of this instrument? You probably meant this to be temporary, working code instead of something you wish to commit definitely to the code base (right?) but I would still fix this asap. Hardcoding values like this is a very big no-no. 

Add a method like `load_calibration_data` which takes a file descriptor (not name) as input and stores the calibration parameters as a private class attribute (e.g. `self._calibration_parameters`). The advantage of a file descriptor instead of a name is that it will make you function more easily testable in CI. An example of how this would work is shown here: 

test: https://github.com/QCoDeS/Qcodes/blob/master/qcodes/tests/drivers/test_lakeshore_file_parser.py
driver: https://github.com/QCoDeS/Qcodes/blob/master/qcodes/instrument_drivers/Lakeshore/Model_325.py

Reply:
@sohailc Do you think it is a bad idea to load the calibration by just supplying the filename? If yes, I'll change the code.